Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -771,8 +771,13 @@ extension TaskHandler: ChannelDuplexHandler {
                                    uri: request.uri)
         var headers = request.headers
 
-        if !request.headers.contains(name: "Host") {
-            headers.add(name: "Host", value: request.host)
+        if !request.headers.contains(name: "host") {
+            let port = request.port
+            var host = request.host
+            if (port != 80 || request.scheme != "http") && (port != 443 || request.scheme != "https") {
```

Comment:
random nit: I personally would find this easier to read after de-morganing this once to

    if ! ((port == 80 && request.scheme == "http") || (port == 443 || request.scheme == "https")) {

but I guess that's personal preference. So please feel totally free to ignore.

Reply:
I’m happy to change

Reply:
@adam-fowler pick what _you_ find easier to read :)

Reply:
I actually prefer the middle ground of 
```
if !(port == 80 && request.scheme == "http") && !(port == 443 && request.scheme == "https") 
```

Reply:
although swift format still changes the && into a ,

Reply:
:( silly swift format